odoo/models.py

```diff
@@ -4506,6 +4518,61 @@ def _parent_store_update(self):
         self.env.cache.update(records, self._fields['parent_path'], updated.values())
         records.modified(['parent_path'])
 
+    def _identify_existing_records(self, data_list, search_fields):
```

Comment:
1) maybe replace `_identify_existing_records` by "_identify_unique_matching_records"

- remove `existing`: all records queried in the database by definition exists so its redundant
- add `unique`: you throw an error when it's not unique
- add `matching`: for expression of intents

as it's something added to the `data_list` it would be nice to express it in the name too